### PR TITLE
Remove fp-ts-contrib

### PR DESCRIPTION
**Pull request by XVincentX**
fp-ts 2.4 has built-in support for `bind` notation — allowing us to remove `fp-ts-contrib` and refactor some code.

### CODE REVIEW

File: packages/core/src/factory.ts [hunk 1]
```diff
@@ -62,18 +62,18 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
...
     validations: IPrismDiagnostic[]
   ): TE.TaskEither<Error, ResourceAndValidation & { output: Output }> => {
+    const mockCall = components.mock({
+      resource,
+      input: { data, validations },
+      config: config.mock,
+    });
+
     const produceOutput = isProxyConfig(config)
       ? components.forward(
```
Comment (karol-maciaszek) [inline packages/core/src/factory.ts:72]: If the intention was to increase readability then I'd vote on doing also `const forwardCall = ...`. Or leaving both. as they were.

File: packages/http/src/mocker/callback/callbacks.ts [hunk 2]
```diff
@@ -93,13 +85,15 @@ function assembleBody(request?: IHttpOperationRequest): O.Option<{ body: string;
...
-    O.mapNullable(request => request.headers),
+const assembleHeaders = (request?: IHttpOperationRequest, bodyMediaType?: string): O.Option<Dictionary<string>> =>
+  pipe(
+    O.fromNullable(request?.headers),
     O.chain(params =>
       traverseOption(params, param =>
-        doOption.bind('value', generateHttpParam(param)).return(({ value }) => [param.name, value])
+        pipe(
+          generateHttpParam(param),
+          O.map(value => [param.name, value])
```
Comment (karol-maciaszek) [inline packages/http/src/mocker/callback/callbacks.ts:95]: Good catch

File: packages/http/src/validator/index.ts [hunk 3]
```diff
@@ -96,21 +96,20 @@ const findResponseByStatus = (responses: IHttpOperationResponse[], statusCode: n
...
-              isMatch(parsedMediaType.parameters, parsedSelectedContentMediaType.parameters)
-            );
-          })
-        )
+    O.fromNullable(mediaType),
+    O.map(contentType.parse),
+    O.chain(parsedMediaType =>
+      pipe(
+        contents,
+        findFirst(c => {
```
Comment (karol-maciaszek) [inline packages/http/src/validator/index.ts:104]: Totally minor, but if we're around then maybe:
```suggestion
        A.findFirst(c => {
```

File: packages/http/src/utils/runtimeExpression.ts [hunk 4]
```diff
@@ -111,16 +108,14 @@ export function resolveRuntimeExpression(
...
-        .bind('body', O.fromNullable(body))
-        .bind(
-          'path',
-          pipe(
-            lookup(2, parts),
-            O.chain(part => O.tryCatch(() => pointerToPath('#' + part)))
-          )
+      O.fromNullable(body),
+      O.bindTo('body'),
+      O.bind('path', _ =>
```
Comment (karol-maciaszek) [inline packages/http/src/utils/runtimeExpression.ts:113]:
```suggestion
      O.bind('path', () =>
```

Not sure here, but I feel that this argument is commonly skipped so there is no reason to emphasize that it exists.

File: packages/http/src/utils/runtimeExpression.ts [hunk 5]
```diff
@@ -73,11 +72,9 @@ export function resolveRuntimeExpression(
 
   function tryRequestQuery() {
     return pipe(
-      doOption
-        .do(isPart(1, 'query'))
-        .bind('query', O.fromNullable(request.url.query))
-        .bind('part', lookup(2, parts))
-        .done(),
+      isPart(1, 'query'),
+      O.bind('query', _ => O.fromNullable(request.url.query)),
```
Comment (karol-maciaszek) [inline packages/http/src/utils/runtimeExpression.ts:76]: Same as below

**Review by karol-maciaszek — APPROVED** [review]
Minor comments, one lib less, good stuff! :)

File: packages/http/src/validator/index.ts (hunk 3 above)
Comment (XVincentX) [reply packages/http/src/validator/index.ts:104]: 13f478e

File: packages/http/src/utils/runtimeExpression.ts (hunk 5 above)
Comment (XVincentX) [reply packages/http/src/utils/runtimeExpression.ts]: ca7fccd

File: packages/core/src/factory.ts (hunk 1 above)
Comment (XVincentX) [reply packages/core/src/factory.ts:72]: 2297a1b

**Pull request merged**